src/main/java/HelloJavaMesh.java

```diff
@@ -1,76 +1,78 @@
-/*
- * Hello Java RightMesh
- */
-
-import io.left.rightmesh.id.MeshID;
+import io.left.rightmesh.id.MeshId;
 import io.left.rightmesh.mesh.JavaMeshManager;
 import io.left.rightmesh.mesh.MeshManager;
 
 import java.io.Console;
-import java.lang.Thread; 
 import java.util.HashSet;
 
 import static io.left.rightmesh.mesh.MeshManager.DATA_RECEIVED;
 import static io.left.rightmesh.mesh.MeshManager.PEER_CHANGED;
 import static io.left.rightmesh.mesh.MeshManager.REMOVED;
 
+/*
+ * "Hello RightMesh" for Java - a simple application example for using RightMesh in pure Java.
+ */
 public class HelloJavaMesh  {
-    
     private static final int HELLO_PORT = 9876;
-    JavaMeshManager mm;
-    HashSet<MeshID> users;
-    
+    private JavaMeshManager mm;
+    private HashSet<MeshId> users;
+
+    public HelloJavaMesh() {
+        users = new HashSet<>();
+    }
+
+    /**
+     * Entry method. Creates a new instance of the app and starts it.
+     * @param args Command line arguments.
+     */
     public static void main(String[] args) {
-        System.out.println("Starting up the mesh.");
-        
-        try {
-            HelloJavaMesh app = new HelloJavaMesh();
-        } catch(Exception ex) {
-            System.out.println("Something went wrong with mesh: " 
-                + ex.toString());
-        }
+        HelloJavaMesh app = new HelloJavaMesh();
+        app.start();
     }
-    
-    public HelloJavaMesh() throws Exception {
-        users = new HashSet<>();
-        
-        //start up a non-superpeer right mesh
-        mm = new JavaMeshManager(false);
-        
-        //bind the mesh port
-        mm.bind(HELLO_PORT);
-        
-        //set the handling functions (requires java8 or a lambda backport
-        //like retrolamba (otherwise see the android hello mesh to see
-        //how to do it without any of these)
-        mm.on(PEER_CHANGED, this::handlePeerChanged);
-        mm.on(DATA_RECEIVED, this::handleDataReceived);
-        
-        //try to get user input to send messages to all other peers
-        System.out.println("Enter messages to send to all other peers");
+
+    /**
+     * App logic - starts and configures RightMesh, then transmits every string the user enters
```

Comment:
maybe add - to all other devices running either helloJavaMesh or helloMesh (android).